text/0000-rust-governance.md

```diff
@@ -0,0 +1,728 @@
+have extensive implementation experience.
+
+Some example RFCs that fall into this area:
+
+* [Associated types and multidispatch](https://github.com/rust-lang/rfcs/pull/195)
+* [DST coercions](https://github.com/rust-lang/rfcs/pull/982)
+* [Trait-based exception handling](https://github.com/rust-lang/rfcs/pull/243)
+* [Rebalancing coherence](https://github.com/rust-lang/rfcs/pull/1023)
+* [Integer overflow](https://github.com/rust-lang/rfcs/pull/560) (this has high
+  overlap with the library subteam)
+* [Sound generic drop](https://github.com/rust-lang/rfcs/pull/769)
+
+### Library team
+
+Oversees both `std` and, ultimately, other crates in the `rust-lang` github
+organization. The focus up to this point has been the standard library, but we
+will want "official" libraries that aren't quite `std` territory but are still
+vital for Rust. (The precise plan here, as well as the long-term plan for `std`,
+is one of the first important areas of debate for the subteam.) Also includes
+API conventions.
+
+Some example RFCs that fall into this area:
+
+* [Collections reform](https://github.com/rust-lang/rfcs/pull/235)
+* [IO reform](https://github.com/rust-lang/rfcs/pull/517/)
+* [Debug improvements](https://github.com/rust-lang/rfcs/pull/640)
+* [Simplifying std::hash](https://github.com/rust-lang/rfcs/pull/823)
+* [Conventions for ownership variants](https://github.com/rust-lang/rfcs/pull/199)
+
+### Compiler team
+
+Focuses on compiler internals, including implementation of language
+features. This broad category includes work in codegen, factoring of compiler
+data structures, type inference, borrowck, and so on.
+
+There is a more limited set of example RFCs for this subteam, in part because we
+haven't generally required RFCs for this kind of internals work, but here are two:
+
+* [Non-zeroing dynamic drops](https://github.com/rust-lang/rfcs/pull/320) (this
+  has high overlap with language design)
+* [Incremental compilation](https://github.com/rust-lang/rfcs/pull/594)
+
+### Tooling and infrastructure team
+
+Even more broad is the "tooling" subteam, which at inception is planned to
+encompass every "official" (rust-lang managed) non-`rustc` tool:
+
+* rustdoc
+* rustfmt
+* Cargo
+* crates.io
+* CI infrastructure
+* Debugging tools
+* Profiling tools
+* Editor/IDE integration
+* Refactoring tools
+
+It's not presently clear exactly what tools will end up under this umbrella, nor
+which should be prioritized.
+
+### Moderation team
+
+Finally, the moderation team is responsible for dealing with CoC violations.
+
+One key difference from the other subteams is that the moderation team does not
+have a leader. Its members are chosen directly by the core team, and should be
+community members who have demonstrated the highest standard of discourse and
+maturity. To limit conflicts of interest, **the moderation subteam should not
+include any core team members**. However, the subteam is free to consult with
```

Comment:
I think mentioning that the mod subteam should not include core team members is enough for the scope of this RfC;  subteam can internally decide how to pick users beyond that. I would suggest that they avoid current employees though (at least for now), since at the moment there is a strong core-team/Mozilla relation and conflict of interest is possible.